ipspot/functions.py

```diff
@@ -3,13 +3,58 @@
 import argparse
 import ipaddress
 import socket
-from typing import Union, Dict, Tuple, Any
+from typing import Union, Dict, List, Tuple, Any
 import requests
+from requests.adapters import HTTPAdapter
+from urllib3.poolmanager import PoolManager
 from art import tprint
 from .params import REQUEST_HEADERS, IPv4API, PARAMETERS_NAME_MAP
 from .params import IPSPOT_OVERVIEW, IPSPOT_REPO, IPSPOT_VERSION
 
 
+class IPv4HTTPAdapter(HTTPAdapter):
+    """A custom HTTPAdapter that enforces the use of IPv4 for DNS resolution during HTTP(S) requests using the requests library."""
+
+    def init_poolmanager(self, connections: int, maxsize: int, block: bool = False, **kwargs: dict) -> None:
+        """
+        Initialize the connection pool manager using a temporary override of socket.getaddrinfo to ensure only IPv4 addresses are used.
+
+        :param connections: the number of connection pools to cache
+        :param maxsize: the maximum number of connections to save in the pool
+        :param block: whether the connections should block when reaching the max size
+        :param kwargs: additional keyword arguments for the PoolManager
+        """
+        self.poolmanager = PoolManager(
+            num_pools=connections,
+            maxsize=maxsize,
+            block=block,
+            socket_options=self._ipv4_socket_options(),
+            **kwargs
+        )
+
+    def _ipv4_socket_options(self) -> list:
+        """
+        Temporarily patches socket.getaddrinfo to filter only IPv4 addresses (AF_INET).
+
+        :return: an empty list of socket options; DNS patching occurs here
+        """
+        original_getaddrinfo = socket.getaddrinfo
+
+        def ipv4_only_getaddrinfo(*args: list, **kwargs: dict) -> List[Tuple]:
+            results = original_getaddrinfo(*args, **kwargs)
+            return [res for res in results if res[0] == socket.AF_INET]
+
+        self._original_getaddrinfo = socket.getaddrinfo
+        socket.getaddrinfo = ipv4_only_getaddrinfo
+
+        return []
```

Comment:
I can't understand this function. You're basically rewriting the `socket.getaddrinfo` with a new function. why you're returning an empty list?
Also what would happen if this function is called more than once, it shouldn't right?